code/Player/Controllers/HumanController.cs

```diff
@@ -0,0 +1,676 @@
+﻿using Sandbox;
+using Input = Sandbox.Input;
+
+namespace ssl.Player.Controllers
+{
+	[Library]
+	public class HumanController : BasePlayerController
+	{
+		public float SprintSpeed { get; set; } = 320.0f;
+		public float WalkSpeed { get; set; } = 150.0f;
+		public float DefaultSpeed { get; set; } = 190.0f;
+		public float Acceleration { get; set; } = 10.0f;
+		public float AirAcceleration { get; set; } = 50.0f;
+		public float GroundFriction { get; set; } = 4.0f;
+		public float StopSpeed { get; set; } = 100.0f;
+		public float DistEpsilon { get; set; } = 0.03125f;
+		public float GroundAngle { get; set; } = 46.0f;
+		public float StepSize { get; set; } = 18.0f;
+		public float MaxNonJumpVelocity { get; set; } = 140.0f;
+		public float BodyGirth { get; set; } = 32.0f;
+		public float BodyHeight { get; set; } = 72.0f;
+		public float EyeHeight { get; set; } = 64.0f;
+		public float Gravity { get; set; } = 800.0f;
+		public float AirControl { get; set; } = 30.0f;
+		public bool Swimming { get; set; } = false;
+		public bool AutoJump { get; set; } = false;
+		
+		public HumanController()
+		{
+		}
+
+		/// <summary>
+		/// This is temporary, get the hull size for the player's collision
+		/// </summary>
+		public override BBox GetHull()
+		{
+			var girth = BodyGirth * 0.5f;
+			var mins = new Vector3( -girth, -girth, 0 );
+			var maxs = new Vector3( +girth, +girth, BodyHeight );
+
+			return new BBox( mins, maxs );
+		}
+
+
+		// Duck body height 32
+		// Eye Height 64
+		// Duck Eye Height 28
+
+		protected Vector3 mins;
+		protected Vector3 maxs;
+
+		public virtual void SetBBox( Vector3 mins, Vector3 maxs )
+		{
+			if ( this.mins == mins && this.maxs == maxs )
+				return;
+
+			this.mins = mins;
+			this.maxs = maxs;
+		}
+
+		/// <summary>
+		/// Update the size of the bbox. We should really trigger some shit if this changes.
+		/// </summary>
+		public virtual void UpdateBBox()
+		{
+			var girth = BodyGirth * 0.5f;
+
+			var mins = new Vector3( -girth, -girth, 0 ) * Pawn.Scale;
+			var maxs = new Vector3( +girth, +girth, BodyHeight ) * Pawn.Scale;
+
+			SetBBox( mins, maxs );
+		}
+
+		protected float SurfaceFriction;
+
+
+		public override void FrameSimulate()
+		{
+			base.FrameSimulate();
+
+			EyeRot = Input.Rotation;
+		}
+
+		public override void Simulate()
```

Comment:
What is the diff between simulate and framesimulate?

Reply:
Simulate = Logic
Frame Simulate = Graphics